workspaces/ui-v2/src/optic-components/PromptNavigateAway.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import { FC, useEffect } from 'react';
+import { Prompt } from 'react-router';
+
+type PromptNavigateAwayProps = {
+  shouldPrompt: boolean;
+};
+
+export const PromptNavigateAway: FC<PromptNavigateAwayProps> = ({
```

Comment:
There's two types of prompts (browser based and javascript based)
Browser prompt (closing the tab / page unload (typing a new url into the browser + navigating))
<img width="1754" alt="Screen Shot 2021-04-27 at 10 12 49 AM" src="https://user-images.githubusercontent.com/18374483/116283806-35aa5400-a741-11eb-9e9b-e82ec0be763f.png">

React router navigation (clicking the docs/diff tab)
<img width="1760" alt="Screen Shot 2021-04-27 at 10 12 56 AM" src="https://user-images.githubusercontent.com/18374483/116283888-4eb30500-a741-11eb-9b29-7b6a5dae3da5.png">

Reply:
nice